### DISCUSSION TITLE
micropython machine library

### DISCUSSION THREAD

**Discussion by Emmanuel789** [micropython/micropython#10824] [ESP32]
Hi all!

I am preparing to program ESP32 with micropython, and after trying to install machine library, which I was using few years earlier I received an error that it's impossible to install pycrypt package. Is there some solution for this issue, or maybe alternatives for machine library to control GPIO pins?

**Comment by bixb922** [comment]
Did you try to install a package called "machine"? That package is already installed in Micropython, no need to install. It includes Pin and ADC classes to work with GPIO pins.
What pcrypt library did you try to install? Is it a Python library or  Micropython library? There are Micropython specific cryptography packages available, but I am not familiar with those, for example https://docs.micropython.org/en/latest/library/cryptolib.html

**Reply by Emmanuel789** [reply]
Thank you for your reply.
Just like you said, machine package was already installed and I can use it.
I don't know why I had such issue, maybe it was interpreting my code as normal python where it doesn't exist.
